### PR TITLE
New blog post on retain on delete

### PR DESCRIPTION
**Pull request by Frassle**

### CODE REVIEW

File: themes/default/content/blog/retainondelete/index.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+option](https://github.com/pulumi/pulumi/issues/7747) to tell the Pulumi engine to not actually delete a
+resource when replacing or removing it from the stack.
+
+## Retain on delete
+
+We now support a new resource option [retain on delete]({{< relref "docs/intro/concepts/resources/options/retainOnDelete" >}})
+that allows you to specifiy this case to the engine.
+
+When a resource marked with `retainOnDelete` is deleted or replaced by a Pulumi update the engine will not
+call through to the resource providers `Delete` function. This is the similar to AWS Cloud Formation's [Retain
```
Comment (mikhailshilkov) [inline themes/default/content/blog/retainondelete/index.md:55]: Maybe the engine-providers relationship is too much details here. I would say that it doesn't delete the actual resource in the cloud.

File: themes/default/content/blog/retainondelete/index.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+DeletionPolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html).
+
```
Comment (mikhailshilkov) [inline themes/default/content/blog/retainondelete/index.md:57]: Consider adding a link sentence explaining that the following few sections are going to be about scenario examples.

File: themes/default/content/blog/retainondelete/index.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+## Supporting data storage policies
+
+Using this new feature it is now possible to uncouple resource lifecycles from deployment cycles. This can be
+espically useful to teams with policy around data and storage. For example your company may require that all
```
Comment (mikhailshilkov) [inline themes/default/content/blog/retainondelete/index.md:61]:
```suggestion
especially useful to teams with policy around data and storage. For example your company may require that all
```

File: themes/default/content/blog/retainondelete/index.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+S3 storage is backed up and retained for some amount of time. Using `RetainOnDelete` you can ensure that the
+S3 buckets that Pulumi creates aren't deleted by Pulumi, leaving the data alive for your storage policy team
+to backup and then delete when no longer required.
+
+## Supporting shared resources
+
+Another workflow this option enables is the ability to manage externaly owned resources via Pulumi deploments.
```
Comment (mikhailshilkov) [inline themes/default/content/blog/retainondelete/index.md:68]:
```suggestion
Another workflow this option enables is the ability to manage externally owned resources via Pulumi deployments.
```

Consider running the draft through a spell checker.

File: themes/default/content/blog/retainondelete/index.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+You may have an external long lived resource that you want to temporarily manage via a Pulumi program, but
+then relquinsh back to the external owner rather than deleting when your finished with it.
+
+Using a combiniation of [import](https://www.pulumi.com/docs/guides/adopting/import/) and `retainOnDelete` you
+can bring a resource under managment and then relinqish it back to the external owner.
+
+## Working around provider bugs
```
Comment (mikhailshilkov) [inline themes/default/content/blog/retainondelete/index.md:75]: Should we sugarcoat "bugs"? Sounds a bit harsh :)

File: themes/default/content/blog/retainondelete/index.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+## Retain on delete
+
+We now support a new resource option [retain on delete]({{< relref "docs/intro/concepts/resources/options/retainOnDelete" >}})
+that allows you to specifiy this case to the engine.
+
+When a resource marked with `retainOnDelete` is deleted or replaced by a Pulumi update the engine will not
+call through to the resource providers `Delete` function. This is the similar to AWS Cloud Formation's [Retain
+DeletionPolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html).
+
+## Supporting data storage policies
```
Comment (mikhailshilkov) [inline themes/default/content/blog/retainondelete/index.md:58]:
```suggestion
### Supporting data storage policies
```

Use cases could be subsections

**Review by mikhailshilkov — COMMENTED** [review]
My 2c

File: themes/default/content/blog/retainondelete/index.md (hunk 5 above)
Comment (Frassle) [reply themes/default/content/blog/retainondelete/index.md:75]: problems, issues, failures?

File: themes/default/content/blog/retainondelete/index.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    - fraser-waters
+
+# At least one tag is required. Lowercase, hyphen-delimited is recommended.
+tags:
+    - features
+
+# See the blogging docs at https://github.com/pulumi/docs/blob/master/BLOGGING.md.
+# for additional details, and please remove these comments before submitting for review.
+---
+
```
Comment (lukehoban) [inline themes/default/content/blog/retainondelete/index.md:37]: Slight aside - but reviews on markdown docs in GitHub are quite a bit easier if we don't use hard line breaks for paragraphs.  Then edits can be suggested within a paragraph in a consistent way.  We generally don't use hard line breaks in our hugo site content in part for this reason.

File: themes/default/content/blog/retainondelete/index.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+While Pulumi lets you manage your infrastructure with code there are some cases where operations need to
```
Comment (lukehoban) [inline themes/default/content/blog/retainondelete/index.md:38]: This sounds a little "negative".  I wonder if we can reframe as "Pulumi is frequently used to manage the entire lifecycle of a resource, from creation, to updates, to replacement, to deletion.  However, there are some cases where it is important to ensure that a resource's life can extend *beyond* the lifetime of the Pulumi program that created it.  To support these use cases, Pulumi now support a new resource option `RetainOnDelete` which allows a resource to be retained in the backing cloud provider even after it is delete from the Pulumi stack it is part of."

File: themes/default/content/blog/retainondelete/index.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+option](https://github.com/pulumi/pulumi/issues/7747) to tell the Pulumi engine to not actually delete a
+resource when replacing or removing it from the stack.
+
+## Retain on delete
+
+We now support a new resource option [retain on delete]({{< relref
+"docs/intro/concepts/resources/options/retainOnDelete" >}}) that allows you to specify this case to the
+engine.
+
+When a resource marked with `retainOnDelete` is deleted or replaced by a Pulumi update the engine it doesn't
```
Comment (lukehoban) [inline themes/default/content/blog/retainondelete/index.md:55]:
```suggestion
When a resource marked with `retainOnDelete` is deleted or replaced by a Pulumi update, the resource is removed from the Pulumi state but Pulumi does not
```

File: themes/default/content/blog/retainondelete/index.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+delete the actual resource in the cloud. That is it won't call through to the resource providers `Delete`
```
Comment (lukehoban) [inline themes/default/content/blog/retainondelete/index.md:56]:
```suggestion
delete the actual resource in the cloud. That is it won't call through to the resource provider's `Delete`
```

File: themes/default/content/blog/retainondelete/index.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+deployments. You may have an external long lived resource that you want to temporarily manage via a Pulumi
+program, but then relinquish back to the external owner rather than deleting when your finished with it.
+
+Using a combination of [import](https://www.pulumi.com/docs/guides/adopting/import/) and `retainOnDelete` you
+can bring a resource under management and then relinquish it back to the external owner.
+
+### Working around provider issues
+
+Lastly not really a new workflow but `retainOnDelete` can also be used to work around some provider issues.
+Some providers will fail to refresh and correctly detect that a resource is now deleted leading pulumi to
```
Comment (lukehoban) [inline themes/default/content/blog/retainondelete/index.md:84]:
```suggestion
Some providers will fail to refresh and correctly detect that a resource is now deleted leading Pulumi to
```

File: themes/default/content/blog/retainondelete/index.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+continually erroring trying to issue a delete call to something that no longer exists. While you could always
```
Comment (lukehoban) [inline themes/default/content/blog/retainondelete/index.md:85]:
```suggestion
error trying to issue a delete call to something that no longer exists. While you could always
```

File: themes/default/content/blog/retainondelete/index.md (hunk 7 above)
Comment (Frassle) [reply themes/default/content/blog/retainondelete/index.md:37]: Ah sorry, that's my default for vscode to handle that we do want hard breaks in our code files. I can toggle it off for these posts.

**Review by susanev — APPROVED** [review]

File: themes/default/content/blog/retainondelete/index.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+meta_desc: Introducing the new resource option RetainOnDelete.
+meta_image: meta.png
+authors:
+    - fraser-waters
+tags:
+    - features
+
+---
+
+Pulumi is frequently used to manage the entire lifecycle of a resource, from creation, to updates, to replacement, to deletion. However, there are some cases where it is important to ensure that a resource's life can extend beyond the lifetime of the Pulumi program that created it. To support these use cases, Pulumi now support a new resource option `RetainOnDelete` which allows a resource to be retained in the backing cloud provider even after it is delete from the Pulumi stack it is part of.
```
Comment (lblackstone) [inline themes/default/content/blog/retainondelete/index.md:13]:
```suggestion
Pulumi is frequently used to manage the entire lifecycle of a resource, from creation, to updates, to replacement, to deletion. However, there are some cases where it is important to ensure that a resource's life can extend beyond the lifetime of the Pulumi program that created it. To support these use cases, Pulumi now support a new resource option `RetainOnDelete` which allows a resource to be retained in the backing cloud provider even after it is deleted from the Pulumi stack it is part of.
```

**Pull request merged**